cmd/amp/main.go

```diff
@@ -27,279 +15,39 @@ var (
-
-	whoAmICmd = &cobra.Command{
-		Use:     "whoami",
-		Short:   "Display currently logged-in user",
-		Example: " ",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return whoAmI()
-		},
-	}
-
-	logoutCmd = &cobra.Command{
-		Use:     "logout",
-		Short:   "Logout current user",
-		Example: " ",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return logout()
-		},
-	}
-
-	username string
-	password string
+	//address string
+	config *cli.Configuration
 )
 
-func init() {
-	RootCmd.AddCommand(infoCmd)
-
-	RootCmd.AddCommand(helpCmd)
-	RootCmd.AddCommand(loginCmd)
-	RootCmd.AddCommand(switchCmd)
-	RootCmd.AddCommand(whoAmICmd)
-	RootCmd.AddCommand(logoutCmd)
-
-	RootCmd.SetUsageTemplate(usageTemplate)
-	RootCmd.SetHelpTemplate(helpTemplate)
-
-	RootCmd.PersistentFlags().StringVar(&configFile, "use-config", "", "Specify config file (overrides default at $HOME/.config/amp/amp.yaml)")
-	RootCmd.PersistentFlags().BoolVar(&displayConfigFilePath, "config-used", false, "Display config file used (if any)")
-	RootCmd.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Verbose output")
-	RootCmd.PersistentFlags().StringVar(&serverAddr, "server", "", "Server address")
-	RootCmd.PersistentFlags().BoolVarP(&listVersion, "version", "V", false, "Version number")
-	RootCmd.PersistentFlags().BoolP("help", "h", false, "Display help")
-
-	loginCmd.Flags().StringVar(&username, "name", "", "Account Name")
-	loginCmd.Flags().StringVar(&password, "password", "", "Password")
-
-	switchCmd.Flags().StringVar(&username, "name", username, "Account Name")
-}
-
-// All main does is process commands and flags and invoke the app
 func main() {
-	cobra.OnInitialize(func() {
-		cli.InitConfig(configFile, Config, verbose, serverAddr)
-		if addr := RootCmd.Flag("server").Value.String(); addr != "" {
-			Config.AmpAddress = addr
-		}
-		AMP = cli.NewAMP(Config, cli.NewLogger(Config.Verbose))
-		if !Config.Verbose {
-			RootCmd.SilenceErrors = true
-			RootCmd.SilenceUsage = true
-		}
-
-		//initialize command manager
-		mgr = cli.NewCmdManager(false)
-
-		cli.AtExit(func() {
-			if AMP != nil {
-				AMP.Disconnect()
-			}
-		})
-	})
-
-	cmd, _, err := RootCmd.Find(os.Args[1:])
-	if err != nil {
-		mgr.Fatal(grpc.ErrorDesc(err))
-	}
-	if err := cmd.Execute(); err != nil {
-		mgr.Fatal(grpc.ErrorDesc(err))
-	}
-	cli.Exit(0)
-}
-
-// login validates the input command line arguments and allows login to an existing account
-// by invoking the corresponding rpc/storage method
-func login(amp *cli.AMP, cmd *cobra.Command) error {
-	if cmd.Flag("name").Changed {
-		username = cmd.Flag("name").Value.String()
-	} else {
-		fmt.Print("username: ")
-		username = getName()
-	}
-	if cmd.Flag("password").Changed {
-		password = cmd.Flag("password").Value.String()
-	} else {
-		password = getPassword()
+	// Initial configuration (before processing environment and flags)
+	config = &cli.Configuration{
+		Version: Version,
+		Build:   Build,
+		Address: "m1:50101",
```

Comment:
Sure. I had host and port before. But maybe we can hold off on updating this until we actually need to.